pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1019,4 +1019,14 @@ self: super: {
       { patches = (drv.patches or []) ++ [ patch ];
         editedCabalFile = null;
       });
+
+  # https://github.com/haskell/cabal/issues/4969
+  haddock-library_1_4_4 = dontHaddock
+    (super.haddock-library_1_4_4.override
+       { # The 'attoparsec' haddock-library depends on is its own
```

Comment:
This issue is fixed in 40a3d775a5887ccc944d009032f70f0dd2b891f7. `cabal2nix-1.7.1` no longer generates the bogus dependency.